pyfixest/estimation/feiv_.py

```diff
@@ -193,3 +206,263 @@ def get_fit(self) -> None:
         # Compute bread matrix
         D = np.linalg.inv(self._tXZ @ self._tZZinv @ self._tZX)
         self._bread = H.T @ D @ H
+
+    def first_stage(self) -> None:
+        """Implement First stage regression."""
+        from pyfixest.estimation.estimation import feols
+
+        #  Start First Stage
+        _X = self._X
+        _Z = self._Z
+        _Y = self._Y
+
+        # Store names of instruments from Z matrix
+        self._non_exo_instruments = list(set(self._coefnames_z) - set(self._coefnames))
+
+        # Prepare your data
+        # Select the columns from self._data that match the variable names
+        data = pd.DataFrame(self._Z_1st_stage, columns=self._coefnames_z)
+
+        # Store instrument variable matrix for future use
+        _Z_iv = data[self._non_exo_instruments]
+        self._Z_iv = _Z_iv.to_numpy()
+        # Dynamically create the formula string
+        # Extract names of fixed effect
+
+        data["Y"] = self._endogvar_1st_stage
+        independent_vars = " + ".join(
+            data.columns[:-1]
+        )  # All columns except the last one ('Y')
+
+        # Set fix effects, cluster, and weight options to be passed to feols.
+
+        if self._has_fixef:
+            FE_vars = self._fixef.split("+")
+            data[FE_vars] = self._data[FE_vars]
+            formula = f"Y ~ {independent_vars} | {self._fixef}"
+        else:
+            formula = f"Y ~ {independent_vars}"
+
+        # Type hint to reflect that vcov_detail can be either a dict or a str
+        vcov_detail: Union[dict[str, str], str]
+
+        if self._is_clustered:
+            a = self._clustervar[0]
+            vcov_detail = {self._vcov_type_detail: a}
+            data[a] = self._data[a]
+        else:
+            vcov_detail = self._vcov_type_detail
+
+        if self._has_weights:
+            data["weights"] = self._weights
+            weight_detail = "weights"
+        else:
+            weight_detail = None
+
+        # Do first stage regression
+
+        model1 = feols(
+            fml=formula,
+            data=data,
+            vcov=vcov_detail,
+            weights=weight_detail,
+            collin_tol=1e-10,
+        )
+
```

Comment:
This check is only to avoid that the first stage model might be an IV model itself?

Reply:
This is actually to avoid `mypy` error. If I do not force `model1` to be the `Feols` class like this, it looks like mypy also checks whether these attributes are defined in `FixestMulti` class, and outputs the error flags like the followings; 

```mypy
> git -c user.useConfigOnly=true commit --quiet
debug statements (python)................................................Passed
trim trailing whitespace.................................................Passed
fix end of files.........................................................Passed
check yaml...........................................(no files to check)Skipped
check toml...........................................(no files to check)Skipped
check for added large files..............................................Passed
Validate GitHub Workflows............................(no files to check)Skipped
ruff.....................................................................Passed
ruff-format..............................................................Passed
nbqa-ruff............................................(no files to check)Skipped
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

pyfixest/estimation/feiv_.py:286: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_beta_hat"  [union-attr]
pyfixest/estimation/feiv_.py:289: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_Y_hat_link"  [union-attr]
pyfixest/estimation/feiv_.py:292: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_u_hat"  [union-attr]
pyfixest/estimation/feiv_.py:434: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_k"  [union-attr]
pyfixest/estimation/feiv_.py:460: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "wald_test"  [union-attr]
pyfixest/estimation/feiv_.py:461: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_f_statistic"  [union-attr]
pyfixest/estimation/feiv_.py:462: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_p_value"  [union-attr]
pyfixest/estimation/feiv_.py:500: error: Item "FixestMulti" of "Feols | FixestMulti" has no attribute "_vcov"  [union-attr]
Found 8 errors in 1 file (checked 1 source file)

```